drivers/media/i2c/imx708.c

```diff
@@ -1515,6 +1521,10 @@ static int imx708_start_streaming(struct imx708 *imx708)
 		return ret;
 	}
 
+	/* Quad Bayer re-mosaic adjustments (for full-resolution mode only) */
+	imx708_write_reg(imx708, 0xC428, IMX708_REG_VALUE_08BIT, !qbc_adjust);
+	imx708_write_reg(imx708, 0xC429, IMX708_REG_VALUE_08BIT, qbc_adjust);
```

Comment:
0xc428 is an enable and 0xc429 is a strength?

Checking datasheet it is, but with 0xc428 = 0 being enabled, and = 1 being disabled. Quirky!

I'd be tempted to use `qbc_adjust ? 0x01 : 0x00` instead of `!qbc_adjust` to make it more obvious the values being programmed, otherwise you end up checking the C specification to confirm that !0 = 1. Even better would be macros defining the bits in the register.

Reply:
I've seen many examples of ! and !! for this elsewhere but I agree ?: is clearer.

Reply:
Just make sure you get them the right way round....

Reply:
Now reworked and hopefully clearer, if a bit less terse.